src/overlays/actors/ovl_En_Owl/z_en_owl.c

```diff
@@ -40,137 +71,1139 @@ const ActorInit En_Owl_InitVars = {
+                                this->unk_409 = (s32)(Rand_ZeroFloat(15.0f) + 5.0f);
+                                this->unk_408 = 2;
+                            }
+                            break;
+
+                        case 2:
+                            sp36 = 0x1388;
+                            if (this->unk_409 <= 0) {
+                                this->unk_408 = 3;
+                                this->unk_409 = 4;
+                            }
+                            break;
+
+                        case 3:
+                            sp36 = Math_SinS(this->unk_409 << 0xC) * 5000.0f;
+                            if (this->unk_409 <= 0) {
+                                this->unk_40A = (s32)Rand_ZeroFloat(20.0f) + 60;
+                                this->unk_408 = 0;
+                                func_8095AB4C(this);
+                            }
+                            break;
+
+                        case 4:
+                            sp36 = Math_SinS(this->unk_409 << 0xD) * 5000.0f;
+                            if (this->unk_409 <= 0) {
+                                this->unk_40A = (s32)Rand_ZeroFloat(20.0f) + 60;
+                                this->unk_408 = 0;
+                                func_8095AB4C(this);
+                            }
+                            break;
+                    }
+
+                    if (this->actionFlags & 0x20) {
+                        sp36 = -sp36;
+                    }
+                }
+
+                if (this->unk_40D > 0) {
+                    this->unk_40D--;
+                } else {
+                    this->unk_40C--;
+                    switch (this->unk_40B) {
+                        case 0:
+                            this->unk_3DE = (-this->unk_40C * 0x5DC) + 0x1770;
+                            if (this->unk_40C <= 0) {
+                                this->unk_40B = 1;
+                                this->unk_40C = (s8)(Rand_ZeroFloat(15.0f) + 5.0f);
+                            }
+                            break;
+
+                        case 1:
+                            this->unk_3DE = 0x1770;
+                            if (this->unk_40C <= 0) {
+                                this->unk_40B = 2;
+                                this->unk_40C = 4;
+                            }
+                            break;
+
+                        case 2:
+                            this->unk_3DE = this->unk_40C * 0x5DC;
+                            if (this->unk_40C <= 0) {
+                                this->unk_40B = 0;
+                                this->unk_40C = 4;
+                                this->unk_40D = (s32)Rand_ZeroFloat(40.0f) + 160;
+                            }
+                            break;
+                    }
+                }
+            }
+
+            if (sp36) {};
+            this->unk_3DC = (u16)((this->unk_3DA << 2) << 8) + sp36;
```

Comment:
Does this need to be two separate shifts? Could it be one, or one or two multiplications?

Reply:
It's a single << 0xA but there's regalloc/code gen problems if I don't use 2. I was stuck on this and copied the idea from OoT: https://github.com/zeldaret/oot/blob/master/src/overlays/actors/ovl_En_Owl/z_en_owl.c#L1256 

There must be some macro going on here I think.